src/main/java/org/mozilla/zest/core/v1/ZestAssignFromElement.java

```diff
@@ -0,0 +1,251 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+package org.mozilla.zest.core.v1;
+
+import net.htmlparser.jericho.Element;
+import net.htmlparser.jericho.Source;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class ZestAssignFromElement extends ZestAssignment {
+
+    private boolean filterElementName;
+    private String elementNameFilter;
+
+    private boolean filterAttributeValue;
+    private String attributeNameFilter;
+    private String attributeValueFilter;
+    private int attributeValueFilterFlags;
+    transient private Pattern attributeValueFilterPattern;
+
+    private boolean reverseFilteredElements;
+    private int elementIndex;
+
+    private boolean returnElement;
+
+    private boolean returnAttribute;
+    private String returnedAttributeName;
+
+    public ZestAssignFromElement() {
+    }
+
+    public ZestAssignFromElement(String variableName) {
+        super(variableName);
+    }
+
+    @Override
+    public String assign(ZestResponse response, ZestRuntime runtime) throws ZestAssignFailException {
+        if (response == null) {
+            throw new ZestAssignFailException(this, "Null response");
+        }
+
+        Source src = new Source(response.getBody());
+        List<Element> elementsFilteredByElementName = filterByElementNameIfConfigured(src);
+        List<Element> elementsFilteredByAttributeValue = filterByAttributeValueIfConfigured(elementsFilteredByElementName);
+        List<Element> elements = reverseIfConfigured(elementsFilteredByAttributeValue);
+        Element element = findElementAtIndex(elements);
+        if(element == null){
+            return null;
```

Comment:
I think it might be better that the consumer of the Assign handle it in the way he likes.
```
if(result == null){
 doSomethingElse()
}
```
Maybe with a `ZestExpressionEquals`. Though ZAP can not compare with `null` at the moment.

If I throw an exception at this point the whole zest pipeline is aborted, right? And that is not my expected behaviour.